github4s/shared/src/main/scala/github4s/api/PullRequests.scala

```diff
@@ -28,87 +28,94 @@ import scala.language.higherKinds
-   *   - base: Filter pulls by base branch name. Example: `gh-pages`.
-   *   - sort: What to sort results by. Can be either `created`, `updated`, `popularity` (comment count)
-   *     or `long-running` (age, filtering by pulls updated in the last month). Default: `created`
-   *   - direction: The direction of the sort. Can be either `asc` or `desc`.
-   *     Default: `desc` when sort is created or sort is not specified, otherwise `asc`.
-   * @return a GHResponse with the pull request list.
-   */
+    * List pull requests for a repository
+    *
+    * @param accessToken to identify the authenticated user
+    * @param headers optional user headers to include in the request
+    * @param owner of the repo
+    * @param repo name of the repo
+    * @param filters define the filter list. Options are:
+    *   - state: Either `open`, `closed`, or `all` to filter by state. Default: `open`
+    *   - head: Filter pulls by head user and branch name in the format of `user:ref-name`.
+    *     Example: `github:new-script-format`.
+    *   - base: Filter pulls by base branch name. Example: `gh-pages`.
+    *   - sort: What to sort results by. Can be either `created`, `updated`, `popularity` (comment count)
+    *     or `long-running` (age, filtering by pulls updated in the last month). Default: `created`
+    *   - direction: The direction of the sort. Can be either `asc` or `desc`.
+    *     Default: `desc` when sort is created or sort is not specified, otherwise `asc`.
+    * @return a GHResponse with the pull request list.
+    */
   def list(
-      accessToken: Option[String] = None,
-      headers: Map[String, String] = Map(),
-      owner: String,
-      repo: String,
-      filters: List[PRFilter] = Nil): M[GHResponse[List[PullRequest]]] =
+    accessToken: Option[String] = None,
+    headers: Map[String, String] = Map(),
+    owner: String,
+    repo: String,
+    filters: List[PRFilter] = Nil,
+    pagination: Option[Pagination] = None): M[GHResponse[List[PullRequest]]] =
     httpClient.get[List[PullRequest]](
       accessToken,
       s"repos/$owner/$repo/pulls",
       headers,
-      filters.map(_.tupled).toMap)
+      filters.map(_.tupled).toMap,
+      pagination)
 
   /**
-   * List files for a specific pull request
-   *
-   * @param accessToken to identify the authenticated user
-   * @param headers optional user headers to include in the request
-   * @param owner of the repo
-   * @param repo name of the repo
-   * @param number of the pull request for which we want to list the files
-   * @return a GHResponse with the list of files affected by the pull request identified by number.
-   */
+    * List files for a specific pull request
+    *
+    * @param accessToken to identify the authenticated user
+    * @param headers optional user headers to include in the request
+    * @param owner of the repo
+    * @param repo name of the repo
+    * @param number of the pull request for which we want to list the files
```

Comment:
Could you please add the parameter description here?

Reply:
Done.